e2e-tests/portal/cypress/e2e/pages/dashboardDeals.js

```diff
@@ -3,14 +3,42 @@ const page = {
   tableHeader: (column) => cy.get(`[data-cy="deal__header--${column}"]`),
   rows: () => cy.get('.govuk-table__body .govuk-table__row'),
   exporterButton: () => cy.get('[data-cy="deal__header--exporter-button"]'),
+
+  // NOTE (TODO: Add documentation)
+
+  // * selector is "containing"
+  // ^ selector is "starts with"
+  // if these are working as expected, update rowIndex to use *, to be consistent
   row: {
-    exporter: (id) => cy.get(`[data-cy="deal__exporter--${id}"]`),
-    bankRef: (id) => cy.get(`[data-cy="deal__bankRef--bankInternalRefName${id}"]`),
-    link: (id) => cy.get(`[data-cy="deal__link--${id}"]`),
-    product: (id) => cy.get(`[data-cy="deal__product--${id}"]`),
-    status: (id) => cy.get(`[data-cy="deal__status--${id}"]`),
-    type: (id) => cy.get(`[data-cy="deal__submissionType--${id}"]`),
-    updated: (id) => cy.get(`[data-cy="deal__updated--${id}"]`),
+    exporter: (id) => cy.get(`[data-cy*="deal__exporter--${id}"]`),
+    bankRef: (id) => cy.get(`[data-cy*="deal__bankRef--${id}"]`),
+    link: (id) => cy.get(`[data-cy*="deal__link--${id}"]`),
+    product: (id) => cy.get(`[data-cy*="deal__product--${id}"]`),
+    status: (id) => cy.get(`[data-cy*="deal__status--${id}"]`),
+    type: (id) => cy.get(`[data-cy*="deal__submissionType--${id}"]`),
+    updated: (id) => cy.get(`[data-cy*="deal__updated--${id}"]`),
+  },
+  rowIndex: {
+    exporter: (index = 1) => cy.get(`[data-cy*="deal__exporter--index--${index}"]`),
+    // exporter: (index = 1) => cy.get('.govuk-table__body .govuk-table__row').eq(index).find('td').eq(0),
+
+    link: (index = 1) => cy.get(`[data-cy*="deal__link--index--${index}"]`),
+    // link: (index = 1) => cy.get('.govuk-table__body .govuk-table__row').eq(index).find('a'),
```

Comment:
Please remove all the commented out code.

Reply:
done